corehq/apps/users/models.py

```diff
@@ -526,13 +526,24 @@ class Meta(object):
 
 class IsMemberOfMixin(DocumentSchema):
 
-    def _is_member_of(self, domain):
-        return domain in self.get_domains() or (
-            self.is_global_admin() and
-            not domain_restricts_superusers(domain)
-        )
+    def _is_member_of(self, domain, include_enterprise):
+        if self.is_global_admin() and not domain_restricts_superusers(domain):
+            return True
 
-    def is_member_of(self, domain_qs):
+        domains = self.get_domains()
+        if domain in domains:
+            return True
+
+        if include_enterprise:
+            for domain in domains:
```

Comment:
why loop over all domains, rather than call for the domain passed in?

Reply:
5131064418f